MM-63447/MM-65101 Remaining changes for deprecated registerPostDropdownMenuComponent API

hmhealey opened the pull request:
#### Summary
These are a bit late, but they're small enough that I'm comfortable squeezing them in before feature complete tomorrow. That's also why I'm comfortable jamming them together into a single PR. 😅

This PR does the following:
1. Adds a one-time console warning for each plugin using `registerPostDropdownMenuComponent`
2. Fixes the vertical alignment of icons when using `registerPostDropdownMenuAction`

Screenshots of those can be seen below

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-63447
https://mattermost.atlassian.net/browse/MM-65101

#### Screenshots

<img width="647" height="118" alt="Screenshot 2025-09-18 at 2 20 08 PM" src="https://github.com/user-attachments/assets/2b836253-33e4-49b0-996d-3c4111568805" />

|Current, old version of Playbooks|Current, new version of Playbooks|Branch, new version of Playbooks|
-|-|-
<img width="292" height="197" alt="1 - old api" src="https://github.com/user-attachments/assets/ae2c3aa0-81d0-4957-be95-d903c396ad0d" />|<img width="292" height="197" alt="2- new api, current" src="https://github.com/user-attachments/assets/f47a37a5-2142-4779-b732-c9a505cc474a" />|<img width="292" height="197" alt="3 - new api, pr" src="https://github.com/user-attachments/assets/baf67499-902b-4dad-b03d-63eaa5a03697" />

#### Release Note
```release-note
Added console warning when a plugin uses the now-deprecated registerPostDropdownMenuComponent API
```

lieut-data reviewed (approved):
FWIW, feature complete was last Friday: it's judgment day tomorrow :p -- but I'll defer to @amyblais.

yasserfaraazkhan reviewed (approved):
<img width="846" height="531" alt="image" src="https://github.com/user-attachments/assets/86a28e3a-0553-4557-8069-38d5a8655308" />

mattermost-build commented:
Will try to auto merge this PR once all tests and checks are passing. This might take up to an hour.

mattermost-build commented:
Trying to auto merge this PR.

mattermost-build commented:
Cherry pick is scheduled.

hmhealey commented:
I've added `@deprecated` to the method in https://github.com/mattermost/mattermost/pull/34094 which should be picked up by the dev docs next time the [this section](https://developers.mattermost.com/integrate/reference/webapp/webapp-reference/#registry) is generated from the web app code, and I've added the same comment to the type defintions in https://github.com/mattermost/mattermost-plugin-starter-template/pull/232
